gst/nnstreamer/tensor_query/tensor_query_client.c

```diff
@@ -727,7 +727,8 @@ gst_tensor_query_client_chain (GstPad * pad,
   data_h = g_async_queue_timeout_pop (self->msg_queue,
       self->timeout * G_TIME_SPAN_MILLISECOND);
   if (data_h) {
-    self->requested_num--;
+    if (self->requested_num > 0)
+      self->requested_num--;
     ret = nns_edge_data_get_count (data_h, &num_data);
 
     if (ret == NNS_EDGE_ERROR_NONE && num_data > 0) {
```

Comment:
Sorry,
Should code(line 732 ~ 769) be done when ```self->requested_num > 0``` is TRUE ?

Reply:
Actually unnecessary. The param `self->requested_num` is required for limiting max input to query-server.
query-client handles the response from query-server by using msg-queue.